### PR TITLE
fix(uri): created uri helper class to stringify queries

### PR DESCRIPTION
**Pull request by jmy-kellogg**
created uri helper to stringify queries and applied to analytic to avoid undefined errors

BREAKING CHANGE: potential quieres format changes in analytics calls

### CODE REVIEW

File: src/uri-helper.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  public generateUriWithQuery(basePath: string, query?: HandlerQuery): string {
+    let uri
+    if (!query) {
+      uri = `${basePath}`
+    } else if (query.uri || (query.query && query.query.uri)) {
+      uri = query.uri || query.query.uri
+    } else if (query.query) {
+      let flattenQuery = { ...query, ...query.query }
```
Comment (qtotuan) [inline src/uri-helper.ts]: Should we remove the "uri" query param? (also further down below)

File: src/uri-helper.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import qs from 'qs'
+
+export interface HandlerOpiton {
```
[the remark this answers is not in the document]
Comment (jmy-kellogg) [reply src/uri-helper.ts]: done

File: src/uri-helper.ts (hunk 1 above)
Comment (jmy-kellogg) [reply src/uri-helper.ts]: I think the it's unnecessary since the uri is taken care of in the second if conditions.

**Pull request merged**